### PR TITLE
test: migrate logger tests to vitest

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Consolidate the testing frameworks and migrate to vitest.

**Description**

Migrate `logger` tests to vitest.

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/logger/test/e2e/logger/workerLogger.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import fs from "node:fs";
 import worker from "node:worker_threads";
+// eslint-disable-next-line import/no-extraneous-dependencies
 import {expose} from "@chainsafe/threads/worker";
 
 const parentPort = worker.parentPort;
-const workerData = worker.workerData as {logFilepath: string};
+const workerData = worker.workerData;
```
Comment (nazarhussain) [inline packages/logger/test/e2e/logger/workerLogger.js:6]: Have to convert this TS file to JS as `Worker` can only load the JS files. Earlier this was handled by runtime compilation with global node loader.

File: packages/logger/test/unit/node.node.test.ts [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
+            level: LogLevel.info,
+            format,
+            module: opts?.module,
+            timestampFormat: {format: TimestampFormatCode.Hidden},
+          });
           logger.warn(message, context, error);
-          logger.restoreStubs();
-          expect(logger.getLogs()).deep.equals([output[format]]);
+
+          expect((console as TestConsole)._stdout.write).toHaveBeenNthCalledWith(1, `${output[format]}\n`);
```
Comment (nazarhussain) [inline packages/logger/test/unit/node.node.test.ts:32]: The last `\n` added in expectation, which was earlier removed by `stubLoggerForProcessStd`.

File: packages/logger/test/e2e/logger/workerLogger.js [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import fs from "node:fs";
 import worker from "node:worker_threads";
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (nflaig) [inline packages/logger/test/e2e/logger/workerLogger.js]: why is it extraneous? should have `@chainsafe/threads` as dev dependency of logger if it is required for testing

File: packages/logger/test/unit/utils/json.test.ts [hunk 4]
```diff
@@ -27,13 +25,13 @@ describe("Json helper", () => {
 
       // Functions
       // eslint-disable-next-line @typescript-eslint/no-empty-function
-      {id: "function", arg: function () {}, json: "function () { }"},
+      {id: "function", arg: function () {}, json: "function() {\n      }"},
```
Comment (nflaig) [inline packages/logger/test/unit/utils/json.test.ts:28]: I don't really get from where the line breaks are coming

Function toString() does not seem to add them
https://github.com/ChainSafe/lodestar/blob/2530fae45382733f48124fb0afb28084ca4ad60f/packages/logger/src/utils/json.ts#L25-L26

See output
```node
> (function() {}).toString()
'function() {}'
```

Or is this related to what you mentioned here https://github.com/ChainSafe/lodestar/pull/6220#discussion_r1433856180?

Comment (nazarhussain) [reply packages/logger/test/unit/utils/json.test.ts:28]: Yes, the stub earlier was removing the new line break from the output. As no source code is changed related to encoding, so reset assure it behaves it had been before.

Comment (nflaig) [reply packages/logger/test/unit/utils/json.test.ts:28]: But I still don't get why there would be a new line in the middle of the function, what's adding that?

Comment (nazarhussain) [reply packages/logger/test/unit/utils/json.test.ts:28]: When `vitest` transpile the TS code to JS the `esbuild/rollup` add a new line at the function definition, that translates to new line when we do the `function.string()`. 

```js
const arg = function () {
};
console.log(arg.toString());
```

I could not find the direct fix tin the Rollup config for it, and fixing the test fixture is not destructive in this case, so left it like that.

File: packages/test-utils/src/mocha.ts
Comment (nflaig) [inline packages/test-utils/src/mocha.ts:1]: should this file be renamed now?

File: packages/logger/test/unit/utils/json.test.ts (hunk 4 above)
Comment (nflaig) [reply packages/logger/test/unit/utils/json.test.ts:28]: I also don't think it is worth to try and find a fix for it, just good to know why these new lines are there

**Review by nflaig — APPROVED** [review]

File: packages/test-utils/src/mocha.ts
Comment (nazarhussain) [reply packages/test-utils/src/mocha.ts:1]: Will remove it completely when migrate all the tests.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.14.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.14.0) :tada:

**Pull request merged**